src/TensorOneD.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <math.h>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+#include "geopm/Exception.hpp"
+
+#include "TensorOneD.hpp"
+
+namespace geopm
+{
+    TensorOneD::TensorOneD()
+    {
+    }
+
+    TensorOneD::TensorOneD(std::size_t dim)
+    {
+        set_dim(dim);
+    }
+
+    TensorOneD::TensorOneD(const TensorOneD &other)
+    {
+        m_vec = other.m_vec;
+    }
+
+    TensorOneD::TensorOneD(json11::Json input)
+    {
+        set_dim(input.array_items().size());
```

Comment:
This assumes the Json object is an array. I recommend you validate that (I believe Json has an ``is_array()`` function).